dallinger/command_line/docker_ssh.py

```diff
@@ -300,20 +324,32 @@ def deploy(
         "dallinger/Caddyfile",
     )
 
-    print("Removing any pre-existing app with the same name.")
-    app_yml = f"~/dallinger/{app_name}/docker-compose.yml"
-    executor.run(
-        f"if [ -f {app_yml} ]; then docker compose -f {app_yml} down --remove-orphans; fi",
-        raise_=False,
-    )
+    if not update:
+        print("Removing any pre-existing app with the same name.")
+        app_yml = f"~/dallinger/{app_name}/docker-compose.yml"
+        executor.run(
+            f"if [ -f {app_yml} ]; then docker compose -f {app_yml} down --remove-orphans; fi",
+            raise_=False,
+        )
 
-    print("Removing any pre-existing Redis volumes.")
-    remove_redis_volumes(app_name, executor)
+        print("Removing any pre-existing Redis volumes.")
+        remove_redis_volumes(app_name, executor)
+    else:
+        app_yml = f"~/dallinger/{app_name}/docker-compose.yml"
+        yml_file_exists = executor.run(f"ls -l {app_yml}", raise_=False)
```

Comment:
This test doesn't work actually because the docker-compose.yml is currently always uploaded, even in update mode (see above). It would be better to move this test before the docker-compose.yml is uploaded.

Reply:
:facepalm: Indeed! Thanks for pointing it out.

Reply:
Cool, works now!